File: project-euler/problem_3.js
var value = 13195; // 600851475143;

var largest_prime = 0;
var start = 2;

var divide = function (result) {
    var number = result[0];
    var start = result[1];
    for (start; start < number; start++) {
        if (number % n === 0) {
            number = number / n;
            start = n;
            break;
        }
    }

    return [number, start];
};

while (value > 1) {
    result = divide(result);
    largest_prime = result[1];
    value = result[0];
}

alert("Largest Prime: " + largest_prime);

File: project-euler/problem_3.rb
#!/usr/bin/ruby

value = 600851475143

largest_prime = 0
start = 2

def divide(number, start)
  start.upto(number).each do |n|
    if number % n == 0
      number = number / n
      start = n
      break
    end
  end

  puts "#{number}: #{start}"
  return number, start
end

while value > 1 do
  value, start = divide(value, start)
  largest_prime = start
end

puts "largest_prime: #{largest_prime}"
